### PR TITLE
Changing the avatar of the user emits the changeUser event which trig…

### PR DESCRIPTION
**Pull request by DeepDiver1975**
…gers update of the system addressbook

### CODE REVIEW

File: lib/private/avatar.php [hunk 1]
```diff
@@ -39,28 +42,32 @@
...
 		$this->l = $l;
+		$this->user = $user;
 	}
 
 	/**
 	 * get the users avatar
-	 * @param int $size size in px of the avatar, avatars are square, defaults to 64
-	 * @return boolean|\OCP\IImage containing the avatar or false if there's no image
+	 *
+*@param int $size size in px of the avatar, avatars are square, defaults to 64
```
Comment (nickvergessen) [inline lib/private/avatar.php]: woops

**Comment by rullzer** [comment]
@DeepDiver1975 please update the avatar unit tests so they also check if the trigger change function is actually called when then avatar is updated.

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 please update the avatar unit tests so they also check if the trigger change function is actually called when then avatar is updated.

done

**Comment by PVince81** [comment]
Code looks good :+1:

**Comment by rullzer** [comment]
This did not work for me. I updated the avatar (via the webui). But the vcard was not updated. While it was when I changed the displayname.

Also am I correct in assuming there is no avatar info yet in the vcard?

**Comment by DeepDiver1975** [comment]
> Also am I correct in assuming there is no avatar info yet in the vcard?

Avatars are already exported to the vcard - maybe the emitter is not passed into the user class in all scenarios .....

**Comment by DeepDiver1975** [comment]
oh brilliant - some pr recently killed the support for -1 as size to get the original file :imp:

**Comment by DeepDiver1975** [comment]
here we go - https://github.com/owncloud/core/commit/d796c43841f5f22e39af76e2c11d60335e985738#diff-ff9bc8ce9f428c1dab6607dc95bb7dacL82

**Comment by rullzer** [comment]
@DeepDiver1975 ah you want the original image... did that work before? Because before we always tried to resize.. so passing -1 should return an error there I would assume...

anyway... then I would think we would require extra functions to properly return the originial image...

**Comment by DeepDiver1975** [comment]
requires https://github.com/owncloud/core/pull/21885

**Comment by DeepDiver1975** [comment]
@rullzer please retest - THX

**Comment by rullzer** [comment]
Took the liberty to fix the unit tests (tiggerChanged is called twice... once on delete and once of setting the avatar).

Tested with cadaver and works. vcard is updated on delete and on a new avatar.
:+1:

**Comment by DeepDiver1975** [comment]
> Took the liberty to fix the unit tests (tiggerChanged is called twice... once on delete and once of setting the avatar).

for sure - THX

**Comment by PVince81** [comment]
:+1:

**Pull request merged**